ProcessLib/HydroMechanics/HydroMechanicsFEM-impl.h

```diff
@@ -303,6 +303,16 @@ void HydroMechanicsLocalAssembler<ShapeFunctionDisplacement,
                                                           pressure_index)
         .noalias() = -Kup;
 
+    if (_process_data.mass_lumping)
+    {
+        for (int idx_ml = 0; idx_ml < storage_p.cols(); idx_ml++)
+        {
+            double const mass_lump_val = storage_p.col(idx_ml).sum();
+            storage_p.col(idx_ml).setZero();
+            storage_p(idx_ml, idx_ml) = mass_lump_val;
+        }
```

Comment:
No need for the for-loop. It is included in Eigen through the `colwise()` operator. See [RM-FEM](https://github.com/ufz/ogs/blob/master/ProcessLib/RichardsMechanics/RichardsMechanicsFEM-impl.h#L404-L406).
```c++
storage_p = storage_p.colwise().sum().eval().asDiagonal();
```

Reply:
You're absolutely right. I updated the code.